packages/cardano-graphql-services/src/StakePoolSearch/DbSyncStakePoolSearchProvider/mappers.ts

```diff
@@ -1,16 +1,150 @@
+    return Cardano.StakePoolStatus.Active;
+  }
+  if (lastPoolRetirement.retiringEpoch > lastEpoch) return Cardano.StakePoolStatus.Retiring;
+  return Cardano.StakePoolStatus.Retired;
+};
 
-export const mapPoolHash = (poolHashModel: PoolHashModel): PoolHash => ({
-  hash: toHexString(poolHashModel.hash_raw),
-  id: poolHashModel.id,
-  view: poolHashModel.view
+interface ToCoreStakePoolInput {
+  poolOwners: PoolOwner[];
+  poolDatas: PoolData[];
+  poolRegistrations: PoolRegistration[];
+  poolRelays: PoolRelay[];
+  poolRetirements: PoolRetirement[];
+  lastEpoch: number;
+}
+
+export const toCoreStakePool = ({
+  poolOwners,
+  poolDatas,
+  poolRegistrations,
+  poolRelays,
+  poolRetirements,
+  lastEpoch
+}: ToCoreStakePoolInput): Cardano.StakePool[] =>
+  poolDatas.map((poolData) => {
+    const registrations = poolRegistrations.filter((r) => r.hashId === poolData.hashId);
+    const retirements = poolRetirements.filter((r) => r.hashId === poolData.hashId);
+    return {
+      cost: poolData.cost,
+      epochRewards: [],
+      hexId: poolData.hexId,
+      id: poolData.id,
+      // FIXME: this values cannot be taken by db sync
+      margin: {
+        denominator: 1,
+        numerator: 1
+      },
+      metadata: poolData.metadata,
+      metadataJson: poolData.metadataJson,
+      metrics: {} as Cardano.StakePoolMetrics, // TODO: fetch metrics
+      owners: poolOwners.filter((o) => o.hashId === poolData.hashId).map((o) => o.address),
+      pledge: poolData.pledge,
+      relays: poolRelays.filter((r) => r.updateId === poolData.updateId).map((r) => r.relay),
+      rewardAccount: poolData.rewardAccount,
+      status: getPoolStatus(registrations[0], lastEpoch, retirements[0]),
+      transactions: {
+        registration: registrations.map((r) => r.transactionId),
+        retirement: retirements.map((r) => r.transactionId)
+      },
+      vrf: poolData.vrfKeyHash
+    };
+  });
+
+export const mapPoolUpdate = (poolUpdateModel: PoolUpdateModel): PoolUpdate => ({
+  id: poolUpdateModel.id,
+  updateId: poolUpdateModel.update_id
+});
+
+export const mapPoolData = (poolDataModel: PoolDataModel): PoolData => {
+  const vrfAsHexString = toHexString(poolDataModel.vrf_key_hash);
+  const toReturn: PoolData = {
+    cost: BigInt(poolDataModel.fixed_cost),
+    hashId: poolDataModel.hash_id,
+    hexId: Cardano.PoolIdHex(vrfAsHexString),
+    id: Cardano.PoolId(poolDataModel.pool_id),
+    pledge: BigInt(poolDataModel.pledge),
+    rewardAccount: Cardano.RewardAccount(poolDataModel.reward_address),
+    updateId: poolDataModel.update_id,
+    vrfKeyHash: Cardano.VrfVkHex(vrfAsHexString)
+  };
+  if (poolDataModel.metadata_hash) {
+    toReturn.metadataJson = {
+      hash: Cardano.util.Hash32ByteBase16(toHexString(poolDataModel.metadata_hash)),
+      url: poolDataModel.metadata_url
+    };
+  }
+  if (poolDataModel.offline_data) {
+    toReturn.metadata = JSON.parse(poolDataModel.offline_data) as Cardano.StakePoolMetadataFields;
```

Comment:
Do these types match?

Reply:
They do. Still, it has a `json` type so probably we want to add some validations